## Apify for Twitter Customer Sentiment

The Twitter API was deprecated in early 2023, making the functionality of the API very limited. In order to have more access to the API, the basic tier requires $100/month. Rather than utilizing this API, in order to gather data from Twitter, I have employed the use of Apify, an open source service which has a `Twitter Webscraper` that allows free scraping based on search term, Twitter URL, dates, etc, as long as the information is public (not getting data from a private account, etc). For this use case, I simply used the following search terms and hashtags:
- UNIQA
- UNIQA.at
- UNIQA Insurance Group
- #UNIQA
- #UNIQA.at
- #UNIQAInsuranceGroup

**It is important to note that this analysis will most likely be flawed/biased as this service doesn't allow many queries and the Twitter API has been deprecated. I will not be able to get very significant data most likely. 

In [1]:
#import libraries
import pandas as pd

I queried the LATEST posts with 50 requests, but it resulted in an output of only 20 tweets. 

In [3]:
#latest twitter posts mentioning UNIQA
twitter_latest_uniqa = pd.read_csv('dataset_twitter-scraper_2024-04-18_11-53-55-829.csv')
twitter_latest_uniqa

url  \
0   https://twitter.com/LegiaKosz/status/178091439...   
1   https://twitter.com/SIGALUNIQAGROUP/status/178...   
2   https://twitter.com/SIGALUNIQAGROUP/status/178...   
3   https://twitter.com/SIGALUNIQAGROUP/status/178...   
4   https://twitter.com/nova_praha/status/17808389...   
5   https://twitter.com/MitPolska/status/178058976...   
6   https://twitter.com/Cashlesspl/status/17805693...   
7   https://twitter.com/dentalinfo_nl/status/17805...   
8   https://twitter.com/KaVoDental/status/17805368...   
9   https://twitter.com/NowinyLokalne/status/17804...   
10  https://twitter.com/XiioUniqa/status/178039334...   
11  https://twitter.com/NUNBOC/status/178032870618...   
12  https://twitter.com/SIGALUNIQAGROUP/status/178...   
13  https://twitter.com/SIGALUNIQAGROUP/status/178...   
14  https://twitter.com/BloombergAdrBiH/status/178...   
15  https://twitter.com/gazetaua_twit/status/17801...   
16  https://twitter.com/uniqa_cz_sk/status/1780146...   
17  https://twitter.com/QaahGarcia/status/17799145...   
18  https://twitter.com/SIGALUNIQAGROUP/status/177...   
19  https://twitter.com/SIGALUNIQAGROUP/status/177...   

                        created_at  \
0   Thu Apr 18 11:00:54 +0000 2024   
1   Thu Apr 18 09:29:43 +0000 2024   
2   Thu Apr 18 09:29:33 +0000 2024   
3   Thu Apr 18 09:29:16 +0000 2024   
4   Thu Apr 18 06:01:16 +0000 2024   
5   Wed Apr 17 13:30:57 +0000 2024   
6   Wed Apr 17 12:09:48 +0000 2024   
7   Wed Apr 17 11:24:57 +0000 2024   
8   Wed Apr 17 10:00:29 +0000 2024   
9   Wed Apr 17 06:01:07 +0000 2024   
10  Wed Apr 17 00:30:25 +0000 2024   
11  Tue Apr 16 20:13:35 +0000 2024   
12  Tue Apr 16 11:48:57 +0000 2024   
13  Tue Apr 16 11:48:46 +0000 2024   
14  Tue Apr 16 10:00:00 +0000 2024   
15  Tue Apr 16 09:59:02 +0000 2024   
16  Tue Apr 16 08:10:06 +0000 2024   
17  Mon Apr 15 16:47:51 +0000 2024   
18  Mon Apr 15 13:06:22 +0000 2024   
19  Mon Apr 15 13:06:07 +0000 2024   

                                            full_text  favorite_count  \
0   ⛹️Sport to pasja i emocje ale i również trudne...               4   
1   Merr më shumë ℹ️ në: https://t.co/TeR97Hy9dh\n...               0   
2   Ju përfitoni 👇🏻:\n✅Dëmshpërblim në masën e shu...               0   
3   Lufta kundër tumoreve është një prej sfidave m...               0   
4   Počty střetů a havárií skútrů narůstají podle ...               0   
5   Współodpowiedzialność w walce z hejtem - darmo...               1   
6   Uniqa ułatwi zdalne kupowanie polis. Klienci p...               2   
7   KaVo organiseert de Dental Excellence Benelux-...               0   
8   #KaVo #uniQa – Four aspects largely determine ...               0   
9   Rzeczywistość bez przemocy: Uniwersytet SWPS, ...               0   
10                                       E vuelto xd.               0   
11  @qmerty94729 @LukaszGibala No i? Strabag jest ...               0   
12  Me një ekip të specializuar, ne ofrojmë zgjidh...               0   
13  A jeni gati të ndërtoni biznesin tuaj, pa shqe...               0   
14  Još uvijek se borimo s predrasudama da je osig...               1   
15  У спільному спецпроєкті з компанією UNIQA розп...               0   
16  Věděli jste, že k nejnákladnějším pojistným ud...               0   
17  Tengo La Novia Más Tóxica.. Más Hermosa ... Un...               0   
18  ⏩Falë Sigurimit Anti-Tumor ju përfitoni:\n✅Pag...               0   
19  Shqipëria shënon çdo vit rreth 6500 raste të r...               0   

    retweet_count  
0               3  
1               0  
2               0  
3               0  
4               0  
5               0  
6               1  
7               0  
8               0  
9               0  
10              0  
11              0  
12              0  
13              0  
14              0  
15              0  
16              0  
17              0  
18              0  
19              0

I queried the TOP posts with 50 requests, but it resulted in an output of only 20 tweets. This shows how flawed this system probably is.

In [4]:
#top twitter posts mentioning uniqa
twitter_top_uniqa = pd.read_csv('top_dataset_twitter_uniqa.csv')
twitter_top_uniqa

bookmark_count  bookmarked  conversation_id_str  \
0                0       False  1778452962496638990   
1                0       False  1780838991064899757   
2                0       False  1780569346672070786   
3                0       False  1780589767773540446   
4                0       False  1779830650754801724   
5                0       False  1780914396102811734   
6                0       False  1778724347118514255   
7                0       False  1780891336960352296   
8                0       False  1778714084843638814   
9                0       False  1780146636875776382   
10               0       False  1780276382611747136   
11               0       False  1777992571978981555   
12               0       False  1780174293466251354   
13               0       False  1780846812766535972   
14               0       False  1778117516667019295   
15               0       False  1778482764989227277   
16               0       False  1780122605556568124   
17               0       False  1780891336960352296   
18               0       False  1779914545118535939   
19               0       False  1777894356285837315   

                        created_at  entities/hashtags/0/indices/0  \
0   Thu Apr 11 16:00:02 +0000 2024                            NaN   
1   Thu Apr 18 06:01:16 +0000 2024                            NaN   
2   Wed Apr 17 12:09:48 +0000 2024                            NaN   
3   Wed Apr 17 13:30:57 +0000 2024                           86.0   
4   Mon Apr 15 11:14:29 +0000 2024                           21.0   
5   Thu Apr 18 11:00:54 +0000 2024                            NaN   
6   Fri Apr 12 09:58:34 +0000 2024                          117.0   
7   Thu Apr 18 09:29:16 +0000 2024                            NaN   
8   Fri Apr 12 09:17:39 +0000 2024                            NaN   
9   Tue Apr 16 08:10:06 +0000 2024                          238.0   
10  Tue Apr 16 20:13:35 +0000 2024                            NaN   
11  Wed Apr 10 09:30:37 +0000 2024                            NaN   
12  Tue Apr 16 10:00:00 +0000 2024                            NaN   
13  Thu Apr 18 06:32:21 +0000 2024                            NaN   
14  Wed Apr 10 17:47:06 +0000 2024                          110.0   
15  Thu Apr 11 17:58:28 +0000 2024                            NaN   
16  Tue Apr 16 06:34:36 +0000 2024                            NaN   
17  Thu Apr 18 09:29:43 +0000 2024                          104.0   
18  Mon Apr 15 16:47:51 +0000 2024                            NaN   
19  Wed Apr 10 03:00:20 +0000 2024                            0.0   

    entities/hashtags/0/indices/1 entities/hashtags/0/text  \
0                             NaN                      NaN   
1                             NaN                      NaN   
2                             NaN                      NaN   
3                            96.0                PiotrLipa   
4                            25.0                      AIT   
5                             NaN                      NaN   
6                           123.0                    sigal   
7                             NaN                      NaN   
8                             NaN                      NaN   
9                           244.0                    UNIQA   
10                            NaN                      NaN   
11                            NaN                      NaN   
12                            NaN                      NaN   
13                            NaN                      NaN   
14                          115.0                     oevp   
15                            NaN                      NaN   
16                            NaN                      NaN   
17                          110.0                    sigal   
18                            NaN                      NaN   
19                           13.0             KaVoRoadshow   

    entities/hashtags/1/indices/0  entities/hashtags/1/indices/1  \
0                          

In [32]:
#want to see full text without ellipsis (...)
pd.set_option('display.max_colwidth', None)

#top_comments
top_comments = pd.DataFrame(twitter_top_uniqa['full_text'])
top_comments

full_text
0                                                                                                                                                                                                                                                                              My Uniqa iha https://t.co/bpKrdFQkcM
1                                                                                                                                                                                   Počty střetů a havárií skútrů narůstají podle údajů pojišťovny UNIQA každoročně o přibližně 10 procent\nhttps://t.co/C3o1z9KgL2
2                                                                                                                                                                                                 Uniqa ułatwi zdalne kupowanie polis. Klienci potwierdzą tożsamość za pomocą usługi mojeID https://t.co/Y8IKiHkqWn
3        Współodpowiedzialność w walce z hejtem - darmowe wydarzenie 25.04 o 9:00 w Warszawie. #PiotrLipa z #UNIQAPolska podzieli się spostrzeżeniami.\n\nZapisz się: https://t.co/9SmqSl40YB\n\n#wydarzeniaMIT #MIT #MITPolska #UNIQA #hejt #ICANInstitute #ESG #CSR #OdpowiedzialnyBiznes https://t.co/QRUtCP44tT
4                                                                                                                                         No digital barriers! #AIT forces together with #UNIQA the topic digital accessibility at the #CX Forum on 16th and 17th April at Tech Gate Vienna https://t.co/WBveTqNGcB
5      ⛹️Sport to pasja i emocje ale i również trudne momenty. O radzeniu sobie z hejtem w sporcie i dbaniu o przygotowanie psychiczne rozmawiali @wykson91 i Magdalena Chorzewska!👩‍🏫 \n\nSpotkanie odbyło się w ramach projektu "Wrzuć hejt do kosza" realizowanego przez Fundacja UNIQA! https://t.co/H9qDAiX78D
6                                                                                                                  Ose merr më shumë ℹ️: \n\n📲+355 68 606 2829\n📩 info@uniqa.al\n\nMe SIGAL UNIQA Group Austria, jetojmë më mirë së bashku!\n\n#sigal #uniqa #sigaluniqa #jetojmëmëmirësëbashku #25ViteMeSIGALUNIQA
7                                             Lufta kundër tumoreve është një prej sfidave më të mëdha që kërkon jo vetëm spitale dhe mjekë shumë të mirë, por edhe kosto të mëdha financiare 💰\n\nSigurimi Anti-Tumor nga SIGAL UNIQA, është miku juaj 👥 më i mirë në ditët më të vështira https://t.co/xNxhSp4s2k
8         Uniqa ma ambitne plany rozwoju na polskim rynku. Ubezpieczyciel chce sprzedawać więcej polis w salonach samochodowych i rozszerzyć grono sprzedawców. Będzie lepiej promować ofertę i uzupełni ją o pakiety zdrowotne. Więcej na ten temat w PI Finance, fragment tekstu poniżej. https://t.co/0sBpwfW7Ht
9                         Věděli jste, že k nejnákladnějším pojistným událostem z cestovního pojištění patří potíže v Himalájích? ✈️ S ohledem na nadmořskou výšku a vzdálenost od ČR jde vždy o vyšší statisíce či miliony korun. ⛰️ Přečtěte si, co všechno pojištění #UNIQA kryje. ⤵️ \n https://t.co/fnKOVVqP9Y
10                                                                                                                                                                                             @qmerty94729 @LukaszGibala No i? Strabag jest firma austriacką w której udziały też posiada Uniqa Raiffeisen (bank).
11        : A responsible partner and employer on 18 European markets, Austria’s UNIQA Insurance Group has a tradition dating back more than two centuries and today represents a community for a better living comprising over 16 million clients. \n\nRead more - https://t.co/xXR9KayLIb https://t.co/lzAhRhPVj2
12                                                                                                                                                       Još uvijek se borimo s predrasudama da je osiguranje nepotrebno i skupo, kazao je Eldar Dudo, direktor UNIQA osiguranje u BiH 👇\n\nhttps://t.

### Using ChatGPT from OpenAI, I asked to translate these tweets:
**The translations are seen below:**



`Czech` (cs):
Počty střetů a havárií skútrů narůstají podle údajů pojišťovny UNIQA každoročně o přibližně 10 procent
`English`: "The number of scooter collisions and accidents is increasing annually by approximately 10 percent according to UNIQA insurance company data."
- - - - -

`Polish` (pl):
Uniqa ułatwi zdalne kupowanie polis. Klienci potwierdzą tożsamość za pomocą usługi mojeID
`English`: "Uniqa will facilitate remote purchasing of insurance policies. Customers will confirm their identity using the mojeID service."
- - - - -

`Polish` (pl):
Współodpowiedzialność w walce z hejtem - darmowe wydarzenie 25.04 o 9:00 w Warszawie. #PiotrLipa z #UNIQAPolska podzieli się spostrzeżeniami.
`English`: "Co-responsibility in the fight against hate speech - a free event on April 25 at 9:00 am in Warsaw. #PiotrLipa from #UNIQAPolska will share insights."
- - - - -

`English` (en):
No digital barriers! #AIT forces together with #UNIQA the topic digital accessibility at the #CX Forum on 16th and 17th April at Tech Gate Vienna
No translation needed.
- - - - -

`Polish` (pl):
Sport to pasja i emocje ale i również trudne momenty. O radzeniu sobie z hejtem w sporcie i dbaniu o przygotowanie psychiczne rozmawiali @wykson91 i Magdalena Chorzewska!
`English`: "Sport is passion and emotions but also difficult moments. @wykson91 and Magdalena Chorzewska discussed dealing with hate speech in sports and mental preparation!"
- - - - -

`Albanian` (sq):
Ose merr më shumë ℹ️: Me SIGAL UNIQA Group Austria, jetojmë më mirë së bashku!
`English`: "Or get more ℹ️: With SIGAL UNIQA Group Austria, we live better together!"
- - - - -

`Albanian` (sq):
Lufta kundër tumoreve është një prej sfidave më të mëdha që kërkon jo vetëm spitale dhe mjekë shumë të mirë, por edhe kosto të mëdha financiare.
`English`: "The fight against tumors is one of the biggest challenges that requires not only very good hospitals and doctors, but also large financial costs."
- - - - -

`Polish` (pl):
Uniqa ma ambitne plany rozwoju na polskim rynku. Ubezpieczyciel chce sprzedawać więcej polis w salonach samochodowych i rozszerzyć grono sprzedawców.
`English`: "Uniqa has ambitious development plans for the Polish market. The insurer wants to sell more policies in car showrooms and expand the group of sellers."
- - - - -

`Czech` (cs):
Věděli jste, že k nejnákladnějším pojistným událostem z cestovního pojištění patří potíže v Himalájích? S ohledem na nadmořskou výšku a vzdálenost od ČR jde vždy o vyšší statisíce či miliony korun.
`English`: "Did you know that the most expensive insurance events from travel insurance include problems in the Himalayas? Considering the altitude and distance from the Czech Republic, it always involves higher thousands or millions of crowns."
- - - - -

`English` (en):
No translation needed.
- - - - -

`English` (en):
A responsible partner and employer on 18 European markets, Austria’s UNIQA Insurance Group has a tradition dating back more than two centuries and today represents a community for a better living comprising over 16 million clients.
No translation needed.
- - - - -
`Bosnian` (bs):
Još uvijek se borimo s predrasudama da je osiguranje nepotrebno i skupo, kazao je Eldar Dudo, direktor UNIQA osiguranje u BiH
`English`: "We are still fighting prejudices that insurance is unnecessary and expensive, said Eldar Dudo, director of UNIQA Insurance in BiH."
- - - - -
`Polish` (pl):
Pozytywne doświadczenia z Akceleratorem Wiedzy Brokera skłoniły @UNIQAPolska do stworzenia analogicznego programu dla agentów - Akceleratora Wiedzy Agenta. Start przedsięwzięcia już w przyszłym tygodniu
`English`: "Positive experiences with the Broker Knowledge Accelerator prompted @UNIQAPolska to create a similar program for agents - the Agent Knowledge Accelerator. The project will start next week."
- - - - -
`German` (de):
Man kann festhalten: seit dem Maturanten Kanzler mit uniqa Praktikum hat sich das intellektuelle Vermögen der #oevp Politiker nicht mehr gesteigert. Ohne Parteiapparat und Anwälte wären die alle nichts.
`English`: "It can be stated: since the Maturanten Chancellor with a UNIQA internship, the intellectual capacity of the #oevp politicians has not increased. Without the party apparatus and lawyers, they would be nothing."
- - - - -
`Spanish` (es):
Que Nadie Te Apague ... Vs Brilla Xq Brillando SOS Uniqa 💆‍♂️😁
`English`: "Don't Let Anyone Turn You Off ... Shine Because Shining You Are Uniqa 💆‍♂️😁"
- - - - -
`Polish` (pl):
Bartłomiej Biały i Anna Szewczyk z @UNIQAPolska o strategii ubezpieczyciela dotyczącej agentów:
`English`: "Bartłomiej Biały and Anna Szewczyk from @UNIQAPolska about the insurer's strategy regarding agents:"
- - - - -
`Albanian` (sq):
Merr më shumë ℹ️ në: Me SIGAL UNIQA Group Austria, jetojmë më mirë së bashku!
`English`: "Get more ℹ️ at: With SIGAL UNIQA Group Austria, we live better together!"
- - - - -
`Spanish` (es):
Tengo La Novia Más Tóxica.. Más Hermosa ... Uniqa... no La Llega Nadie Es Uniqa No Tieenne Comparación ❤️‍🩹 Te Amo Mi Pichona ❤️😍
`English`: "I have the most toxic girlfriend.. Most beautiful ... Uniqa... nobody compares, Uniqa has no comparison ❤️‍🩹 I love you my little one ❤️😍"
- - - - -
`English` (en):
#KaVoRoadshow 2024 #KaVo is organizing a #roadshow through #Europe starting on April 23rd, presenting the ProXam Imaging portfolio, the KaVo uniQa as well as the latest #dentalinstruments in a traveling showroom.
No translation needed.

## 
I have found that only some of the tweets (about half) actually have anything to do with UNIQA insurance- and the ones that do seem to come only from UNIQA Insurance subsidiaries/locations. None of these tweets are from customers or potential custmer
